cmd/qspi/src/lib.rs

```diff
@@ -532,12 +614,18 @@ fn qspi(context: &mut humility::ExecutionContext) -> Result<()> {
         // Now we're ready to write/verify in units of blocksize.
         //
         let data_size = context.data_size() as u32;
-        let chunk = data_size - (data_size % block_size);
+        let chunk = data_size - (data_size % BLOCK_SIZE);
         let mut offset = 0;
 
         let mut buf = vec![0u8; chunk as usize];
         let mut file = File::open(filename)?;
 
+        // Skip sector0 if the user didn't specify --write-sector0
+        if !subargs.write_sector0 {
+            offset += SECTOR_SIZE;
+            file.seek(std::io::SeekFrom::Start(SECTOR_SIZE as u64)).unwrap();
+        }
```

Comment:
Are we silently doing a partial write of the total requested writes? If so, shouldn't we fail this operation?

Reply:
Yes, unless you pass the `--write-sector0` flag, Humility will skip sector 0 when writing images.  This is added to the documentation at the top of the file.

I've tweaked the logic in ee5a1632431a47f717f7bfec91c0bc59b5daf567: if you write an image that includes anything other than `0xFF` in sector 0, then Humility will stop with an error.  This matches the behavior in `control-plane-agent`, which is motivated by (a) doing the right thing and (b) not requiring more dramatic changes.